signer/src/proto/convert.rs

```diff
@@ -0,0 +1,448 @@
+        }
+    }
+}
+
+impl TryFrom<proto::BitcoinTransactionSignAck> for BitcoinTransactionSignAck {
+    type Error = Error;
+    fn try_from(value: proto::BitcoinTransactionSignAck) -> Result<Self, Self::Error> {
+        Ok(BitcoinTransactionSignAck {
+            txid: BitcoinTxId::try_from(value.txid.required()?)?.into(),
+        })
+    }
+}
+
+impl From<StacksTransactionSignature> for proto::StacksTransactionSignature {
+    fn from(value: StacksTransactionSignature) -> Self {
+        proto::StacksTransactionSignature {
+            txid: Some(StacksTxId::from(value.txid).into()),
+            signature: Some(value.signature.into()),
+        }
+    }
+}
+
+impl TryFrom<proto::StacksTransactionSignature> for StacksTransactionSignature {
+    type Error = Error;
+    fn try_from(value: proto::StacksTransactionSignature) -> Result<Self, Self::Error> {
+        Ok(StacksTransactionSignature {
+            txid: StacksTxId::try_from(value.txid.required()?)?.into(),
+            signature: value.signature.required()?.try_into()?,
+        })
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::keys::PrivateKey;
+
+    use super::*;
+
+    use std::marker::PhantomData;
+
+    use bitcoin::hashes::Hash as _;
+
+    use fake::Dummy;
+    use fake::Fake;
+    use fake::Faker;
+    use rand::rngs::OsRng;
+    use test_case::test_case;
+
+    struct Unit;
+
+    impl Dummy<Unit> for bitcoin::OutPoint {
+        fn dummy_with_rng<R: rand::Rng + ?Sized>(_: &Unit, rng: &mut R) -> Self {
+            let bytes: [u8; 32] = Faker.fake_with_rng(rng);
+            let txid = bitcoin::Txid::from_byte_array(bytes);
+            let vout: u32 = Faker.fake_with_rng(rng);
+            bitcoin::OutPoint { txid, vout }
+        }
+    }
+
+    impl Dummy<Unit> for RecoverableSignature {
+        fn dummy_with_rng<R: rand::Rng + ?Sized>(_: &Unit, rng: &mut R) -> Self {
+            let private_key = PrivateKey::new(rng);
+            let msg = secp256k1::Message::from_digest([0; 32]);
+            private_key.sign_ecdsa_recoverable(&msg)
+        }
+    }
+
+    #[test]
+    fn conversion_between_bytes_and_uint256() {
+        let number = proto::Uint256 {
+            bits_part0: Faker.fake_with_rng(&mut OsRng),
+            bits_part1: Faker.fake_with_rng(&mut OsRng),
+            bits_part2: Faker.fake_with_rng(&mut OsRng),
+            bits_part3: Faker.fake_with_rng(&mut OsRng),
+        };
+
+        let bytes = <[u8; 32]>::from(number);
+        let round_trip_number = proto::Uint256::from(bytes);
+        assert_eq!(round_trip_number, number);
+    }
+
+    #[test]
+    fn conversion_between_uint256_and_bytes() {
+        let bytes: [u8; 32] = Faker.fake_with_rng(&mut OsRng);
+        let number = proto::Uint256::from(bytes);
+
+        let round_trip_bytes = <[u8; 32]>::from(number);
+        assert_eq!(round_trip_bytes, bytes);
+    }
+
+    #[test_case(PhantomData::<([u8; 32], proto::Uint256)>; "Uint256")]
```

Comment:
Yeah we can delete `conversion_between_uint256_and_bytes`, I think the other is still relevant, since it tests the `From<Uint256>` implementation of `[u8; 32]`.

Reply:
Removed the redundant one in https://github.com/stacks-network/sbtc/pull/808/commits/ef0c639d01a55254088f96c4390cda1f840d0bd6.